### PR TITLE
Add support for EBSOptimized option

### PR DESCRIPTION
**Pull request by ans0600**

### CODE REVIEW

File: templates/buildkite-elastic.yml [hunk 1]
```diff
@@ -126,6 +127,11 @@ Parameters:
     Default: t2.nano
     MinLength: 1
 
+  EBSOptimized:
+    Description: Optional - Enabled EBSOptimized for the instance
```
Comment (lox) [inline templates/buildkite-elastic.yml]: "Enabled Optimized EBS for instance types that don't have it automatically enabled" ?

Comment (Vektah) [inline templates/buildkite-elastic.yml]: >  that don't have it automatically enabled" ?

What happens if you specify `EBSOptimized: false` for instances that get it by default? are they unoptimized?

Comment (ans0600) [inline templates/buildkite-elastic.yml]: @lox @Vektah We noticed it will not enable EBSOptimized for you by default. A good example would be m4 family.

Comment (lox) [inline templates/buildkite-elastic.yml]: > The M4 instances are EBS-Optimized by default, with additional, dedicated network capacity for I/O operations. 

Are you saying that's not the case @ans0600?

Comment (ans0600) [inline templates/buildkite-elastic.yml]: @lox try launch a spot m4 instance use this template and see if it is set to true. It is not set for us.

Comment (lox) [inline templates/buildkite-elastic.yml]: @ans0600 are you able to follow up with Amazon and enquire? I'm reluctant to make this change without some more feedback on how it should work.

**Comment by lox** [comment]
Hey, you didn't need to close it :) My concern was that you are passing a false value where there was none previously. 

If you were able to use an `If` and pass `AWS::NoValue` rather than false for when the parameter wasn't provided, would be happy to merge.

**Pull request closed without merging**